lib/google/gax/retry_settings.rb

```diff
@@ -0,0 +1,89 @@
+# Copyright 2019, Google Inc.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+#     * Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above
+# copyright notice, this list of conditions and the following disclaimer
+# in the documentation and/or other materials provided with the
+# distribution.
+#     * Neither the name of Google Inc. nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+module Google
+  module Gax
+    #
+    # @!attribute initial_delay
+    #   @return [Numeric] the initial delay time, in seconds,
+    #     between the completion of the first failed request and the
+    #     initiation of the first retrying request.
+    # @!attribute delay_multiplier
+    #   @return [Numeric] the multiplier by which to increase the
+    #     delay time between the completion of failed requests, and
+    #     the initiation of the subsequent retrying request.
+    # @!attribute max_delay
+    #   @return [Numeric] the maximum delay time, in seconds,
+    #     between requests. When this value is reached,
+    #     +delay_multiplier+ will no longer be used to
+    #     increase delay time.
+    class RetrySettings
```

Comment:
I'm not sure that we really need this class. These could be hash args that are passed to `RetryManager#initialize`, which might make it a little cleaner for custom retry implementations that are not using a timed backoff approach.

Reply:
I agree that we can do better. retry codes, RetrySettings, and RetryManager can probably be squashed into one object.

Reply:
I've thought about this, and I think I have a design that will collapse RetrySettings and RetryManager into a single object. If a user provides a Hash then we will create this new object, or the user can pass an object that implements `#retry?(error)` and `#delay!` methods instead. That should also satisfy giving users more visibility/control of the errors raised and the retry mechanics. Working on this today.

Reply:
This class has been removed.